sunkit_instruments/goes_xrs/goes_chianti_tem.py

```diff
@@ -0,0 +1,317 @@
+    namely, goes_chianti_tem.pro, and it has been tested against that for consistency.
+
+    It now also works for the GOES-16 and -17 data, and for the re-processed netcdf
+    GOES/XRS files for GOES 13-15.
+
+    Also note that this has only been tested on the high resolutions 1s/2s/3s data of GOES/XRS.
+
+    Parameters
+    ----------
+    goes_ts : `~sunpy.timeseries.XRSTimeSeries`
+        The GOES/XRS timeseries containing the data of both the xrsa and xrsb channels (in units of W/m**2).
+    abundance: {``coronal`` | ``photospheric``}, optional
+        Which abundances to use for the calculation, the default is ``coronal``.
+
+    Returns
+    -------
+    output : `~sunpy.timeseries.GenericTimeSeries`
+        Contains the temperature and emission measure calculated from the input ``goes_ts`` time series.
+
+    Example
+    -------
+    >>> from sunpy import timeseries as ts
+    >>> import sunpy.data.sample  # doctest: +REMOTE_DATA
+    >>> from sunkit_instruments import goes_xrs
+    >>> goes_ts = ts.TimeSeries(sunpy.data.sample.GOES_XRS_TIMESERIES) # doctest: +REMOTE_DATA
+    >>> goes_flare = goes_ts.truncate("2011-06-07 06:20", "2011-06-07 07:30") # doctest: +REMOTE_DATA
+    >>> goes_temp_emiss = goes_xrs.calculate_temperature_em(goes_flare) # doctest: +REMOTE_DATA
+
+    Notes
+    -----
+    This function works with both the NOAA-provided netcdf files, for which the data is given in "true" fluxes
+    and with the older FITS files provided by the SDAC, for which the data are scaled to be consistent with GOES-7.
+    The routine determines from the `sunpy.timeseries.XRSTimeSeries` metadata
+    whether the SWPC scaling factors need to be removed (which are present in the FITS data).
+
+    See also: https://hesperia.gsfc.nasa.gov/goes/goes.html#Temperature/Emission%20Measure
+
+    In regards to the re-processed GOES 8-15 data, please refer to the documentation here:
+        * https://www.ncei.noaa.gov/data/goes-space-environment-monitor/access/science/xrs/GOES_1-15_XRS_Science-Quality_Data_Readme.pdf
+    This includes information of the SWPC scaling factors that were applied to prior data (e.g. the FITS data).
+
+    For the GOES-R data please refer to here:
+        * https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l2/docs/GOES-R_XRS_L2_Data_Readme.pdf
+
+    References
+    ----------
+    .. [1] White, S. M., Thomas, R. J., & Schwartz, R. A. 2005,
+        Sol. Phys., 227, 231, DOI: 10.1007/s11207-005-2445-z
+
+    """
+    if not isinstance(goes_ts, ts.XRSTimeSeries):
+        raise TypeError(
+            f"Input time series must be a XRSTimeSeries instance, not {type(goes_ts)}"
+        )
+
+    if goes_ts.observatory is None:
+        raise ValueError(
+            f"The GOES satellite number was not found in the input time series"
+        )
+
+    satellite_number = int(goes_ts.observatory.split("-")[-1])
+    if (satellite_number < 1) or (satellite_number > 17):
```

Comment:
I wonder if we should define these as constant values somewhere.

Reply:
do you mean like define 1 and 17 somewhere and make them variables defined at the top or something?

Reply:
in theory yes, when GOES-18 data is available. But 1 should always be the same. I'll fix it here to make "max_supported_satellite" variable